### PR TITLE
115 achievements admin

### LINKED ISSUE #115: Create Admin achievement
It should be universal endpoint for adding new achievement with any type. It will be used only by admin through some Rest Client (Postman or so).

### PR DESCRIPTION
**Pull request by billMeLater**
Close #115

### CODE REVIEW

File: src/main/java/juja/microservices/gamification/service/AchievementService.java [hunk 1]
```diff
@@ -87,6 +88,52 @@
         return Collections.singletonList(id);
     }
 
+    public List<String> addAdmin(AdminAddRequest request) {
+        List<String> result = new ArrayList<>();
+        AchievementType type = request.getType();
+
+        logger.debug("Received request on /achievements/" + type.toString() + " : {}", request);
```
Comment (VadimDyachenko) [inline src/main/java/juja/microservices/gamification/service/AchievementService.java]: You can use slf4j message format:
`logger.debug("Received request on /achievements/{} : {}", type.toString(), request);`

File: src/main/java/juja/microservices/gamification/service/AchievementService.java [hunk 2]
```diff
@@ -87,6 +88,52 @@
...
+            result = addCodenjoy(codenjoyRequest);
+
+        } else {
+            logger.warn("Unprocessable type {}", type.toString());
+        }
+
+        if (result.isEmpty()) {
+            logger.error("Failed to process request {}", request);
+        } else {
+            logger.info("Added '" + type.toString() + "' achievement, id = {}", result.toString());
```
Comment (VadimDyachenko) [inline src/main/java/juja/microservices/gamification/service/AchievementService.java]: You can use slf4j message format

**Comment by adaleorion** [comment]
@billMeLater could you pay attention to this task

File: src/main/java/juja/microservices/gamification/service/AchievementService.java (hunk 2 above)
Comment (billMeLater) [reply src/main/java/juja/microservices/gamification/service/AchievementService.java]: done

File: src/main/java/juja/microservices/gamification/service/AchievementService.java (hunk 1 above)
Comment (billMeLater) [reply src/main/java/juja/microservices/gamification/service/AchievementService.java]: done

File: src/main/java/juja/microservices/gamification/service/AchievementService.java [hunk 3]
```diff
@@ -87,6 +88,52 @@
         return Collections.singletonList(id);
     }
 
+    public List<String> addAdmin(AdminAddRequest request) {
+        List<String> result = new ArrayList<>();
+        AchievementType type = request.getType();
+
+        logger.debug("Received request on /achievements/{} : {}", type.toString(), request);
+
```
Comment (NikolayNN) [inline src/main/java/juja/microservices/gamification/service/AchievementService.java]: why do you write to logger "Received request on /achievements/{type}" 
if request has daily type, that the url will be /achievements/daily
but actual url was /achievements/admin ?
I think better - ("received admin achievement: {}", request)

Comment (billMeLater) [reply src/main/java/juja/microservices/gamification/service/AchievementService.java]: agree, could be confusing, changed.

**Review by adaleorion — APPROVED** [review]
looks good 👍

**Pull request merged**